pkg/operation/botanist/component/nodelocaldns/nodelocaldns.go

```diff
@@ -0,0 +1,543 @@
+									},
+								},
+								Args: []string{
+									"-localip", //Todo add if else condition
+									nodeLocal,
+									"-conf",
+									"\"/etc/Corefile\"",
+									"-upstreamsvc",
+									"kube-dns-updtream",
+								},
+								SecurityContext: &corev1.SecurityContext{
+									Privileged: pointer.Bool(true),
+								},
+								Ports: []corev1.ContainerPort{
+									{
+										ContainerPort: int32(53),
+										Name:          "dns",
+										Protocol:      corev1.ProtocolUDP,
+									},
+									{
+										ContainerPort: int32(53),
+										Name:          "dns-tcp",
+										Protocol:      corev1.ProtocolTCP,
+									},
+									{
+										ContainerPort: int32(prometheusPort),
+										Name:          "metrics",
+										Protocol:      corev1.ProtocolTCP,
+									},
+								},
+								LivenessProbe: &corev1.Probe{
+									Handler: corev1.Handler{
+										HTTPGet: &corev1.HTTPGetAction{
+											Host: nodeLocal,
+											Path: "/health",
+											Port: intstr.FromInt(8080),
+										},
+									},
+									InitialDelaySeconds: int32(60),
+									TimeoutSeconds:      int32(5),
+								},
+								VolumeMounts: []corev1.VolumeMount{
+									{
+										MountPath: "/run/xtables.lock",
+										Name:      "xtables-lock",
+										ReadOnly:  false,
+									},
+									{
+										MountPath: "/etc/coredns",
+										Name:      "config-volume",
+									},
+									{
+										MountPath: " /etc/kube-dns",
+										Name:      "kube-dns-config",
+									},
+								},
+							},
+						},
+						Volumes: []corev1.Volume{
+							{
+								Name: "xtables-lock",
+								VolumeSource: corev1.VolumeSource{
+									HostPath: &corev1.HostPathVolumeSource{
+										Path: "/run/xtables.lock",
+										Type: &hostPathFileOrCreate,
+									},
+								},
+							},
+							{
+								Name: "kube-dns-config",
+								VolumeSource: corev1.VolumeSource{
+									ConfigMap: &corev1.ConfigMapVolumeSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: "kube-dns",
+										},
+										Optional: pointer.Bool(true),
+									},
+								},
+							},
+							{
+								Name: "config-volume",
+								VolumeSource: corev1.VolumeSource{
+									ConfigMap: &corev1.ConfigMapVolumeSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: "node-local-dns", //is updated below
+										},
+										Items: []corev1.KeyToPath{
+											{
+												Key:  configDataKey,
+												Path: "Corefile.base",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		vpa *autoscalingv1beta2.VerticalPodAutoscaler
+	)
+	utilruntime.Must(kutil.MakeUnique(configMap))
+	daemonset.Spec.Template.Spec.Volumes[2].VolumeSource.ConfigMap.LocalObjectReference.Name = configMap.Name
+	utilruntime.Must(references.InjectAnnotations(daemonset))
```

Comment:
Can you rather move the configMap initialization and its MakeUnique call before the DaemonSet initialization? You can probably check how it is handled for the dependency-watchdog component.